automation/src/test/java/org/greenplum/pxf/automation/features/json/JsonTest.java

```diff
@@ -11,198 +11,291 @@
+		exTable.setFields(tweetsFields);
+		exTable.setUserParameters(new String[]{
+				"IDENTIFIER=created_at",
+				"MAXLENGTH=566"});
+
+		gpdb.createTableAndVerify(exTable);
+
+		// Verify results
+		runTincTest("pxf.features.hdfs.readable.json.exceed_max_size.runTest");
+	}
+
+	/**
+	 * Test JSON file with pretty print format. One of the records
+	 * is malformed. In that case the whole line will be
+	 * replaced by NULLs.
+	 *
+	 * @throws Exception if test fails to run
+	 */
+	@Test(groups = {"features", "gpdb"})
+	public void malFormatedRecord() throws Exception {
+
+		exTable.setName("jsontest_malformed_record");
+		exTable.setProfile("Json");
+		exTable.setPath(hdfsPath + FILENAME_BROKEN + SUFFIX_JSON);
+		exTable.setFields(tweetsFields);
+		exTable.setUserParameters(new String[]{"IDENTIFIER=created_at"});
+
+		gpdb.createTableAndVerify(exTable);
+
+		// Verify results
+		runTincTest("pxf.features.hdfs.readable.json.malformed_record.runTest");
+	}
+
+	/**
+	 * Test JSON file with pretty print format with reject limit configured. One of the records
+	 * is malformed. The query is allowed and a table is created.
+	 *
+	 * @throws Exception if test fails to run
+	 */
+	@Test(groups = {"features", "gpdb"})
+	public void malFormatedRecordWithRejectLimit() throws Exception {
+
+		exTable.setName("jsontest_malformed_record_with_reject_limit");
+		exTable.setProfile("Json");
+		exTable.setPath(hdfsPath + FILENAME_BROKEN + SUFFIX_JSON);
+		exTable.setFields(tweetsFields);
+		exTable.setUserParameters(new String[]{"IDENTIFIER=created_at"});
+		exTable.setSegmentRejectLimit(2);
+
+		gpdb.createTableAndVerify(exTable);
+
+		// Verify results
+		runTincTest("pxf.features.hdfs.readable.json.malformed_record_with_reject_limit.runTest");
+	}
+
+	/**
+	 * Test JSON file with all supported types. Some of the records
+	 * have type mismatches (e.g. an integer entered as '(').
+	 * In that case, the line will be sent to GPDB as TEXT, and we
+	 * expect GPDB to raise a type error.
+	 *
+	 * @throws Exception if test fails to run
+	 */
+	@Test(groups = {"features", "gpdb"})
+	public void mismatchedTypesRecord() throws Exception {
+
+		exTable.setName("jsontest_mismatched_types");
+		exTable.setProfile("Json");
+		exTable.setPath(hdfsPath + FILENAME_MISMATCHED_TYPES + SUFFIX_JSON);
+		exTable.setFields(new String[]{
+				"type_int int",
+				"type_bigint bigint",
+				"type_smallint smallint",
+				"type_float real",
+				"type_double float8",
+				"type_string1 text",
+				"type_string2 varchar",
+				"type_string3 bpchar",
+				"type_char char",
+				"type_boolean bool",
+		});
+
+		gpdb.createTableAndVerify(exTable);
+
+		// Verify results
+		runTincTest("pxf.features.hdfs.readable.json.mismatched_types.runTest");
+	}
+
+	/**
+	 * Test JSON file with all supported types. Some of the records
+	 * have type mismatches (e.g. an integer entered as '(').
+	 * In that case, the line will be sent to GPDB as TEXT, and we
```

Comment:
are we sending the line as text, or as en error?
```suggestion
	 * In that case, the line will be sent to GPDB as TEXT, and we
```

Reply:
We forgot to update comments